apt/minimization/minimizer.py

```diff
@@ -140,24 +160,26 @@ def generalizations(self):
         Return the generalizations derived from the model and test data.
 
         :return: generalizations object. Contains 3 sections: 'ranges' that contains ranges for numerical features,
-                                 'categories' that contains sub-groups of categories for categorical features, and
-                                 'untouched' that contains the features that could not be generalized.
+                 'categories' that contains sub-groups of categories for categorical features, and
+                 'untouched' that contains the features that could not be generalized.
         """
         return self._generalizations
 
     @property
     def ncp(self):
         """
-        Return the NCP score of the generalizations.
+        Return the last calculated NCP score. NCP score is calculated upon calling `fit` (on the training data),
+        `transform' (on the test data) or when explicitly calling `calculate_ncp` and providing it a dataset.
 
-        :return: ncp score as float.
+        :return: NCP score as float.
         """
         return self._ncp
```

Comment:
It's worth having an NCP class, which would return the most appropriate NCP according to the process that was done. Then you won't need the "generalize_using_transform" field to be filled by the user, but only pass it as a parameter to the calculate ncp method, where you know whether transform is being used or not.

Reply:
Added NCPScores class instead of single score.